src/name_bazaar/ui/events/registrar_events.cljs

```diff
@@ -51,6 +52,95 @@
                    :on-tx-receipt [:district0x.snackbar/show-message
                                    (gstring/format "%s was successfully registered" ens-record-name)]}]})))
 
+(reg-event-fx
+  :registrar/start-auction
+  [interceptors (validate-first-arg (s/keys :req [:registrar/label]))]
+  (fn [{:keys [:db]} [form-data]]
+    (let [label (normalize (:registrar/label form-data))
+          ens-record-name (str label constants/registrar-root)]
+      {:dispatch [:district0x/make-transaction
+                  {:name (gstring/format "Open registration for %s" ens-record-name)
+                   :contract-key :registrar
+                   :contract-method :start-auction
+                   :form-id {:registrar/label (:registrar/label form-data)}
+                   :form-data (assoc form-data :registrar/label-hash (sha3 label))
+                   :result-href (path-for :route.ens-record/detail {:ens.record/name ens-record-name})
+                   :args-order [:registrar/label-hash]
+                   :tx-opts {:gas 700000 :gas-price default-gas-price}
+                   :on-tx-receipt [:district0x.snackbar/show-message
+                                   (gstring/format "%s registration successfully opened"
+                                                   ens-record-name)]}]})))
+
+(defn- seal-bid [label-hash owner bid-value bid-salt]
+  (letfn [(strip-0x [s]
```

Comment:
you can define this as util function in district0x library. Avoid using `letfn`, it's not very readable